llvm/tools/llvm-c-test/echo.cpp

```diff
@@ -767,11 +767,9 @@ struct FunCloner {
         int NumIdx = LLVMGetNumIndices(Src);
         for (int i = 1; i <= NumIdx; i++)
           Idx.push_back(CloneValue(LLVMGetOperand(Src, i)));
-        if (LLVMIsInBounds(Src))
-          Dst = LLVMBuildInBoundsGEP2(Builder, ElemTy, Ptr, Idx.data(), NumIdx,
-                                      Name);
-        else
-          Dst = LLVMBuildGEP2(Builder, ElemTy, Ptr, Idx.data(), NumIdx, Name);
+
+        Dst = LLVMBuildGEP2(Builder, ElemTy, Ptr, Idx.data(), NumIdx, Name);
```

Comment:
We should add `LLVMBuildGEPWithFlags` or something instead. This build+set pattern is unsafe if the expression is constant folded.

Reply:
Pushed a new version that does this. In the process I realised it wasn't handling GEP constant expressions, so that is done now as well, along with a test to verify it